### PR TITLE
bpo-39305: Refactor nntplib.py

### PR DESCRIPTION
**Pull request by LucianaMarques**
* Join two classes (NNTP and _NNTPBase)
* Modify one test accordingly

https://bugs.python.org/issue39305

### CODE REVIEW

**Comment by LucianaMarques** [comment]
@corona10 and @vstinner this PR is in reference of [bpo-39305](https://bugs.python.org/issue39305), could you please take a loot? Thanks :)

File: Lib/nntplib.py [hunk 1]
```diff
@@ -310,8 +309,9 @@ class _NNTPBase:
     encoding = 'utf-8'
     errors = 'surrogateescape'
 
-    def __init__(self, file, host,
-                 readermode=None, timeout=_GLOBAL_DEFAULT_TIMEOUT):
+    def __init__(self, host, file=None,
```
Comment (corona10) [inline Lib/nntplib.py:312]: I don't want to add a new parameter `file` for NNTP.

**Review by corona10 — CHANGES_REQUESTED** [review]

**Comment by LucianaMarques** [comment]
Hi, just wanted to register here I have been away for a few days but I'm working on this today.

File: Lib/nntplib.py (hunk 1 above)
Comment (LucianaMarques) [reply Lib/nntplib.py:312]: Ok, I will change this.

**Comment by LucianaMarques** [comment]
Hi there, I'm sorry for the absence here, I will come back to this shortly!

**Comment by corona10** [comment]
@LucianaMarques 
This issue is solved by https://github.com/python/cpython/pull/19817 :)
Thanks for the work, I am looking forward to your future contribution!

**Pull request closed without merging**